### PR TITLE
Fix file upload progress

### PR DESCRIPTION
**Pull request by JeromeDeLeon**
Based on the discussion from [#1096](https://github.com/parse-community/Parse-SDK-JS/issues/1096)

Thanks to @ladigitale for the advice. :+1:

### CODE REVIEW

**Comment by JeromeDeLeon** [comment]
I tested it without `type` and worked just fine. No duplication. Maybe the browser handles the calling of progress behind the scene if it should be calling progress for upload or progress for download.

**Comment by JeromeDeLeon** [comment]
I was thinking the same thing like putting `loaded` and `total` into an object allowing the future values to be wrapped into an object or putting `type` as an object at the end of the parameter but then again, the `type` could possibly remove if we could determine about the collision of `upload progress` and `download progress`. If you could test it on your end, that would be great help :+1:

File: src/RESTController.js [hunk 1]
```diff
@@ -158,15 +158,32 @@ const RESTController = {
...
+            options.progress(type, null);
           }
         }
+      }
+
+      xhr.onprogress = (event) => {
+        handleProgress('download', event);
       };
+
+      if (xhr.upload) {
```
Comment (dplewis) [inline src/RESTController.js:181]: Oops I forgot why this was removed. The type parameter can be removed here and we should be good to go.

**Comment by ladigitale** [comment]
I just found the time to test the pacth.
As expected at the end of the upload with this code you will yet have 2 consecutive callbacks of the same function with different meanings : 

1 14535341 14535341   (meaning end of the upload)
1 165 165 (meaning end of the download sent as one chunk because of the low size of the response)

Thats why aditionnal imformations / api evolution would have been usefull.

however I really thank you for that work.

**Comment by ladigitale** [comment]
I hope I don't seem arrogant while writing. English is not my best part.
:)

**Comment by JeromeDeLeon** [comment]
Seems like `type` is really necessary there to distinguish the difference of the two (upload and download). I'll try to push another PR and let you know about it.

**Pull request merged**